testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/OfflineTokenTest.java

```diff
@@ -829,10 +837,35 @@ private void conductOfflineTokenRequest(String expectedRefreshAlg, String expect
         }
     }
 
-    private void testOfflineSessionExpiration(int idleTime, int maxLifespan, int offset) {
+    private String getOfflineClientSessionUuid(final String userSessionId, final String clientId) {
+        return testingClient.server().fetch(session -> {
+            RealmModel realmModel = session.realms().getRealmByName("test");
+            ClientModel clientModel = realmModel.getClientByClientId(clientId);
+            UserSessionModel userSession = session.sessions().getOfflineUserSession(realmModel, userSessionId);
+            AuthenticatedClientSessionModel clientSession = userSession.getAuthenticatedClientSessionByClient(clientModel.getId());
+            return clientSession.getId();
+        }, String.class);
+    }
+
+    private int countSessions(final String userSessionId, final String clientId, final String clientSessionId) {
```

Comment:
Minor: countSessions is likely a bit misleading name of this method as it doesn't return any "count" of sessions? Could be maybe renamed to something like `getTestSessionsState` (or something even better if you can think of something)?

Reply:
In my defense I'm really counting sessions :smile:. 0 is returned if no user and client session exists, 1 if only user session is present, and 2 if both are found. The client session alone cannot be detected because it needs the user session to be retrieved. I changed the method to `checkIfUserAndClientSessionExist`.